crates/red_knot_wasm/src/lib.rs

```diff
@@ -273,10 +277,12 @@ impl Diagnostic {
     #[wasm_bindgen]
     pub fn display(&self, workspace: &Workspace) -> JsString {
         let config = DisplayDiagnosticConfig::default().color(false);
+        let mut buf = vec![];
         self.inner
-            .display(workspace.db.upcast(), &config)
-            .to_string()
-            .into()
+            .borrow()
+            .print(workspace.db.upcast(), &config, &mut buf)
+            .unwrap();
+        String::from_utf8(buf).unwrap().into()
```

Comment:
You could also change the return type to `Result` without breaking the JS Api if you want to avoid the `unwrap` here

Reply:
Ah, and this is a non-test instance of the `unwrap()` that I previously claimed was only in tests. Nice.

OK, that convinces me to switch back to a `Display` impl, kinda like how the old diagnostics work.

But I'll do that in a follow-up PR if that's okay with you. (Which will fix this too.)

Reply:
Sure